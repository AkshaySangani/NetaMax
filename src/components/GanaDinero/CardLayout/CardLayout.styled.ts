import styled from "styled-components";
import { responsive } from "styled/responsive";

import { Box, Flex } from "@chakra-ui/react";

export const CardWrapper = styled(Flex)`
  flex-direction: column;
  text-align: center;
  max-width: 350px;
  width: 100%;
  margin-bottom: 15px;

  ${responsive.mdPlus} {
    max-width: 520px;
    margin-bottom: 50px;
  }
`;

export const BannerWrapper = styled(Box)`
  background: linear-gradient(#aa019c, #5c0178);
  position: absolute;
  padding: 10px 16px;
  border-radius: 18px;
  align-self: center;
  margin-top: -2px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), inset -22px -10px 71px rgba(116, 0, 0, 0.25);

  ${responsive.mdPlus} {
    padding: 12px 30px;
    border-radius: 20px;
    margin-top: -10px;
  }
`;

export const BannerText = styled(Box)`
  color: #fff;
  font-size: 17px;
  font-weight: 700;

  ${responsive.mdPlus} {
    font-size: 22px;
  }
`;

export const BorderWrapper = styled(Box)`
  border: 1px solid #3056ba;
  padding: 4px;
  border-radius: 13px;
`;

export const GradientWrapper = styled(Flex)`
  border-radius: 12px;
  padding: 35px 8px 0 8px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), inset -22px -10px 71px rgba(116, 0, 0, 0.25);
`;
