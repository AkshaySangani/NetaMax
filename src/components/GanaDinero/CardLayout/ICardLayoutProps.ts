export interface ICardLayoutProps {
  /**
   * The text displayed on the purple banner above the card
   * @type string
   */
  bannerText: string;

  /**
   * Array with string to define the gradient colors
   * @type string[]
   */
  gradientColors?: string[];
}
