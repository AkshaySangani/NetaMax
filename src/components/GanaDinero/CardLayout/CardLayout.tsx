import React, { FC, ReactElement } from "react";

import { Box } from "@chakra-ui/react";
import { ICardLayoutProps } from "./ICardLayoutProps";
import {
  BannerText,
  BannerWrapper,
  BorderWrapper,
  CardWrapper,
  GradientWrapper,
} from "./CardLayout.styled";

/**
 * CardLayout
 * @param {ICardLayoutProps} props component props
 * @returns {ReactElement} the react element to render.
 */
export const CardLayout: FC<ICardLayoutProps> = ({
  children,
  bannerText,
  gradientColors,
}): ReactElement => {
  return (
    <CardWrapper>
      <BannerWrapper>
        <BannerText data-testid="card-layout-banner">{bannerText}</BannerText>
      </BannerWrapper>
      <Box pt="20px">
        <Box h="100%">
          <BorderWrapper>
            <GradientWrapper
              data-testid="card-layout-children"
              background={
                gradientColors
                  ? `radial-gradient(at 50% top, ${gradientColors})`
                  : "radial-gradient(#418edd, #2253c9)"
              }
            >
              {children}
            </GradientWrapper>
          </BorderWrapper>
        </Box>
      </Box>
    </CardWrapper>
  );
};
