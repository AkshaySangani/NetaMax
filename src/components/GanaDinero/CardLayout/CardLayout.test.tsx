import { render, cleanup, screen } from "@testing-library/react";
import { ReactElement } from "react";
import { CardLayout } from "./CardLayout";
import { ICardLayoutProps } from "./ICardLayoutProps";

describe("Render CardLayout Component", () => {
  const props: ICardLayoutProps = {
    bannerText: "Test Banner",
  };

  /**
   * TestChildren test component
   * @returns {ReactElement} the react element to render.
   */
  const TestChildren = (): ReactElement => (
    <div>
      <span data-testid="card-layout-test-children">This is a test</span>
    </div>
  );

  test("CardLayout should render properly without children", () => {
    const { container } = render(<CardLayout {...props} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("card-layout-banner").textContent).toMatch(props.bannerText);
    expect(screen.getByTestId("card-layout-children").childElementCount).toBe(0);
  });

  test("CardLayout should render properly with test children", () => {
    const { container } = render(
      <CardLayout {...props}>
        <TestChildren />
      </CardLayout>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("card-layout-children").childElementCount).toBe(1);
    expect(screen.getByTestId("card-layout-test-children").textContent).toMatch("This is a test");
  });

  afterEach(() => {
    cleanup();
  });
});
