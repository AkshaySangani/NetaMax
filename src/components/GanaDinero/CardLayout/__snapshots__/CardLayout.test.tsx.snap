// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CardLayout Component CardLayout should render properly with test children 1`] = `
<div>
  <div
    class="sc-bdvvtL bvVYQj css-k008qs"
  >
    <div
      class="sc-gsDKAQ kWXsUD css-0"
    >
      <div
        class="sc-dkPtRN egXpyz css-0"
        data-testid="card-layout-banner"
      >
        Test Banner
      </div>
    </div>
    <div
      class="css-7w6khc"
    >
      <div
        class="css-10klw3m"
      >
        <div
          class="sc-hKwDye dlqxDY css-0"
        >
          <div
            class="sc-eCImPb ftCsbL css-18duxr7"
            data-testid="card-layout-children"
          >
            <div>
              <span
                data-testid="card-layout-test-children"
              >
                This is a test
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render CardLayout Component CardLayout should render properly without children 1`] = `
<div>
  <div
    class="sc-bdvvtL bvVYQj css-k008qs"
  >
    <div
      class="sc-gsDKAQ kWXsUD css-0"
    >
      <div
        class="sc-dkPtRN egXpyz css-0"
        data-testid="card-layout-banner"
      >
        Test Banner
      </div>
    </div>
    <div
      class="css-7w6khc"
    >
      <div
        class="css-10klw3m"
      >
        <div
          class="sc-hKwDye dlqxDY css-0"
        >
          <div
            class="sc-eCImPb ftCsbL css-18duxr7"
            data-testid="card-layout-children"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
